node/mock_test.go

```diff
@@ -157,7 +164,14 @@ func (p *mockV0Provider) DeletePod(ctx context.Context, pod *v1.Pod) (err error)
 	}
 
 	p.notifier(pod)
-
+	if p.realNotifier == nil {
+		// The pods reconciliation (GetPodStatus) should be called in under 1 minute
```

Comment:
This will get removed once the notifier uses the passed in pod instead of calling the provider again right?

Reply:
Kind of. When it's being run in "Legacy" mode, it'll still need this.

Reply:
Part of me is almost like, maybe we should just actively desupport legacy. It's really not that hard to use the NotifyPods interface.

Reply:
I agree with you but probably not in this PR.